version: '3'

volumes:
  portainer_data:

services:
  portainer:
    image: portainer/portainer-ce:latest
    ports:
      - 9000:9000
      - 9443:9443
    container_name: portainer
    restart: unless-stopped
    security_opt:
      - no-new-privileges:true
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - portainer_data:/data
      - /etc/localtime:/etc/localtime:ro
      - /home/pi/code/portainer/certs:/certs
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.portainer.entrypoints=web,websecure"
      - "traefik.http.routers.portainer.rule=Host(`portainer.mantisd.cloud`)"
      - "traefik.http.routers.portainer.tls=true"
      - "traefik.http.services.portainer.loadbalancer.server.port=9443"
      - "traefik.http.services.portainer.loadbalancer.server.scheme=https"
      - "traefik.http.routers.portainer.tls.certresolver=cloudflare"
      - "traefik.http.routers.portainer.tls.domains[0].main=portainer.mantisd.cloud"
      - "traefik.http.routers.portainer.service=portainer@docker"
#    command:
#      --sslcert /certs/portainer.pem
#      --sslkey /certs/portainer.private.key
    dns:
      - 10.10.10.25
      - 10.10.10.26

networks:
  default:
    external:
      name: proxy

